packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/HttpLogObfuscatorTests.kt

```diff
@@ -0,0 +1,354 @@
+    }
+
+    private object NullObfuscatorLoggerInspector : RealmLogger {
+
+        override val level: LogLevel = LogLevel.DEBUG
+        override val tag: String = "NullObfuscatorLoggerInspector"
+
+        override fun log(
+            level: LogLevel,
+            throwable: Throwable?,
+            message: String?,
+            vararg args: Any?
+        ) {
+            message?.also {
+                if (it.contains(""""password":"***"""")) {
+                    fail("Password was obfuscated: $it")
+                } else if (it.contains(""""access_token":"***","refresh_token":"***"""")) {
+                    fail("Access/refresh tokens were obfuscated: $it")
+                } else if (it.contains(""""key":"***"""")) {
+                    fail("API key was obfuscated: $it")
+                } else if (it.contains(""""id_token":"***"""")) {
+                    fail("Apple/Google ID tokens were obfuscated: $it")
+                } else if (it.contains(""""accessToken":"***"""")) {
+                    fail("Facebook token was obfuscated: $it")
+                } else if (it.contains(""""authCode":"***"""")) {
+                    fail("Google Auth Code was obfuscated: $it")
+                } else if (it.contains(""""token":"***"""")) {
+                    fail("JWT key was obfuscated: $it")
+                } else if (
+                    it.contains(""""arguments":[***]""") ||
+                    it.contains("BODY START\n***\nBODY END")
+                ) {
+                    fail("Custom function arguments were obfuscated: $it")
+                }
+            }
+        }
+    }
+
+    private enum class Operation {
+        OBFUSCATED_PASSWORD,
+        OBFUSCATED_ACCESS_AND_REFRESH_TOKENS,
+        OBFUSCATED_API_KEY,
+        OBFUSCATED_APPLE_OR_GOOGLE_ID_TOKEN,
+        OBFUSCATED_FACEBOOK,
+        OBFUSCATED_GOOGLE_AUTH_CODE,
+        OBFUSCATED_JWT,
+        OBFUSCATED_CUSTOM_FUNCTION
+    }
+
+    @BeforeTest
+    fun setUp() {
+        channel = Channel(1)
+    }
+
+    private fun initApp(): TestApp {
+        return TestApp(
+            appName = "obfuscator",
+            logLevel = LogLevel.DEBUG,
+            customLogger = ObfuscatorLoggerInspector(channel),
+            initialSetup = { app, service ->
+                initializeDefault(app, service)
+                app.addFunction(TestAppInitializer.FIRST_ARG_FUNCTION)
+                app.addFunction(TestAppInitializer.SUM_FUNCTION)
+                app.addFunction(TestAppInitializer.NULL_FUNCTION)
+            }
+        )
+    }
+
+    @AfterTest
+    fun tearDown() {
+        channel.close()
+
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun nullObfuscator() = runBlocking {
+        app = TestApp(
+            appName = "null-obfuscator",
+            logLevel = LogLevel.DEBUG,
+            builder = { it.httpLogObfuscator(null) },
+            customLogger = NullObfuscatorLoggerInspector,
```

Comment:
Seems a bit backwards. I would suggest doing a logger that just collects the messages and verifies the output afterwards like done with the https://github.com/realm/realm-kotlin/blob/main/packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncedRealmTests.kt#L1262.

Reply:
It sounds better than sending events when the expected message has arrived. I will change it.